src/shared/analytics/analytics.client.ts

```diff
@@ -1,85 +1,107 @@
-import { getUserId } from 'src/background/account/account-helpers.client';
 import { emitter } from 'src/ui/shared/events';
 import { getWalletGroupByAddress } from 'src/ui/shared/requests/getWalletGroupByAddress';
+import { readCachedCurrentAddress } from 'src/ui/shared/user-address/useAddressParams';
 import { HandshakeFailed } from '../errors/errors';
+import { rejectAfterDelay } from '../rejectAfterDelay';
 import { createParams, sendToMetabase } from './analytics';
 import {
   createAddProviderHook,
   initialize as initializeApiV4Analytics,
 } from './api-v4-zerion';
 import {
   getProviderForApiV4,
+  getProviderForMetabase,
   getProviderNameFromGroup,
 } from './shared/getProviderNameFromGroup';
 
-function trackAppEvents() {
+function requestWithTimeout<T>(cb: () => Promise<T>, delay = 500) {
+  try {
+    return Promise.race([cb(), rejectAfterDelay(delay, 'Client analytics')]);
+  } catch {
+    return null;
+  }
+}
+
+function trackAppEvents({
+  getWalletProvider,
+}: {
+  getWalletProvider: (address: string) => Promise<string>;
+}) {
+  async function createWalletProperties() {
+    const address = readCachedCurrentAddress() ?? null;
+    const wallet_provider = address
+      ? await requestWithTimeout(() => getWalletProvider(address))
+      : null;
+    return { wallet_address: address?.toLowerCase(), wallet_provider } as const;
```

Comment:
We should filter all `null` values from the final object for Mixpanel. 
Should we filter wallet_provider: null here?
I don't see that we send this to mixpanel, but may be I miss something

Reply:
Loos like all good!